fairscale/experimental/wgit/signal_sparsity.py

```diff
@@ -52,8 +54,25 @@ class SignalSparsity:
             conv.weight.data = 3d_sparser.get_sst_dst_weight(conv.weight.data)
     """
 
-    def __init__(self) -> None:
-        pass
+    def __init__(
+        self,
+        algo: Algo = Algo.FFT,
+        sst_top_k_dim: Optional[int] = -1,
+        sst_top_k_element: Optional[int] = None,
+        sst_top_k_percent: Optional[float] = 0.1,
```

Comment:
BTW, on line 36 and 45, I meant percent value in [0, 100]. It should be checked. Default value could be 99.9, which means 0.1% non-zeros. I got sparsity definition reversed when I wrote 0.1 above in the comment. Sorry about that.

Also, we should make sure top_k_element > 0 and top_k_percent * dim > 0. I.e. no matter what, we should at least have 1 non-zero element in the resulting sparse vector.

Reply:
@min-xu-ai We can boil down ` top_k_percent * dim > 0`  to just ` top_k_percent > 0` right? Otherwise, we start needing the tensor at config validation.

Reply:
We can do this:

top_k_percent * dim -> top_k_element -> top_k_element > 0

this way, both element and percent should share most of the code.

The >0 part is not a validation. It is a low bound. Which means if user specified 0 or small percent, we will never do topk=0, instead, we will always do topk=1 in those cases. Therefore, it is not a validation but a runtime behavior. the reason for this is that I can't think of any valid use case you want topk =0.